src/main/java/uk/gov/hmcts/reform/blobrouter/services/BlobSignatureVerifier.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.reform.blobrouter.services;
+
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.blobrouter.exceptions.DocSignatureFailureException;
+import uk.gov.hmcts.reform.blobrouter.tasks.processors.BlobProcessor;
+import uk.gov.hmcts.reform.blobrouter.util.ZipVerifiers;
+
+import java.io.ByteArrayInputStream;
+import java.util.zip.ZipInputStream;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+@Component
+public class BlobSignatureVerifier {
+
+    private static final Logger logger = getLogger(BlobProcessor.class);
+
+    private final String signatureAlg;
+    private final String publicKeyDerFilename;
+
+    public BlobSignatureVerifier(
+        @Value("${storage-signature-algorithm}") String signatureAlg,
+        @Value("${public-key-der-file}") String publicKeyDerFilename
```

Comment:
personally would like for those to be under some flag. e.g. `storage` as first param suggests. but not sure what we think. plus it's not essential here to discuss :)

Reply:
I'll update it in a separate pr